auth-api/src/auth_api/services/org.py

```diff
@@ -77,45 +78,78 @@ def create_org(org_info: dict, user_id,  # pylint: disable=too-many-locals, too-
         org_info['typeCode'] = OrgType.PREMIUM.value if bcol_account_number else OrgType.BASIC.value
 
         is_staff_admin = token_info and 'staff_admin' in token_info.get('realm_access').get('roles')
-        if not is_staff_admin:  # staff can create any number of orgs
-            count = OrgModel.get_count_of_org_created_by_user_id(user_id)
-            if count >= current_app.config.get('MAX_NUMBER_OF_ORGS'):
-                raise BusinessException(Error.MAX_NUMBER_OF_ORGS_LIMIT, None)
-            if org_info.get('accessType', None) == AccessType.ANONYMOUS.value:
-                raise BusinessException(Error.USER_CANT_CREATE_ANONYMOUS_ORG, None)
+        is_bceid_user = token_info and token_info.get('loginSource', None) == LoginSource.BCEID.value
+
+        Org.validate_account_limit(is_staff_admin, user_id)
+
+        access_type = Org.validate_access_type(is_bceid_user, is_staff_admin, org_info)
 
         if not bcol_account_number:  # Allow duplicate names if premium
             Org.raise_error_if_duplicate_name(org_info['name'])
 
         org = OrgModel.create_from_dict(camelback2snake(org_info))
+        org.access_type = access_type
+        # If the account is anonymous set the billable value as False else True
+        org.billable = access_type in (AccessType.EXTRA_PROVINCIAL.value, AccessType.REGULAR.value)
+
+        # Set the status based on access type
+        # Check if the user is APPROVED else set the org status to PENDING
+        if access_type == AccessType.EXTRA_PROVINCIAL.value \
+                and not AffidavitModel.find_approved_by_user_id(user_id=user_id):
+            org.status_code = OrgStatus.PENDING.value
         org.add_to_session()
 
-        if is_staff_admin:
-            org.access_type = AccessType.ANONYMOUS.value
-            org.billable = False
-        else:
-            org.access_type = AccessType.BCSC.value
-            org.billable = True
-
         # If mailing address is provided, save it
         if mailing_address:
             Org.add_contact_to_org(mailing_address, org)
 
         # create the membership record for this user if its not created by staff and access_type is anonymous
-        if not is_staff_admin and org_info.get('access_type') != AccessType.ANONYMOUS:
+        Org.create_membership(access_type, is_staff_admin, org, user_id)
+
+        Org.add_payment_settings(org.id, bcol_account_number, bcol_user_id)
+
+        org.save()
```

Comment:
not sure if you handled it somwhere , have to send  mail to bcol group when somebody creates an extrapro org rite

Reply:
That ticket is not in progress.